oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/query/ElasticsearchIndex.java

```diff
@@ -48,7 +48,7 @@
     private final ElasticsearchConnection elasticsearchConnection;
     private final NodeState root;
 
-    ElasticsearchIndex(@NotNull ElasticsearchConnection elasticsearchConnection, @NotNull NodeState root) {
+    protected ElasticsearchIndex(@NotNull ElasticsearchConnection elasticsearchConnection, @NotNull NodeState root) {
```

Comment:
This doesn't seem to be used outside package. Can we revert the change to access modifier.

Reply:
Correct. I reverted it and changed most of the class modifiers as well. Most of them, in fact, should be `package private`. I came to this conclusion looking at Java API code and watching this [video](https://www.youtube.com/watch?v=KrLFs6f2bOA) (the first 20 min).

A nice feature of this is the following: in one look you understand what a single package exposes:
<img width="410" alt="Screenshot 2020-03-31 at 16 01 17" src="https://user-images.githubusercontent.com/1469849/78037064-5439f200-736b-11ea-8ddf-d942c6361d65.png">